### PR TITLE
Convert react dashboard

### PR DESCRIPTION
**Pull request by taylorpwhipp**
This PR converts the `react-dashboard` example to Typescript with Next.js.  

The existing react-dashboard will need to be updated to nextjs-dashboard in liveblocsk.io.
PR here:
https://github.com/liveblocks/liveblocks.io/pull/1106/files

Github Issue:
https://github.com/liveblocks/liveblocks.io/issues/1070

### CODE REVIEW

**Review by nvie — COMMENTED** [review]
As I mentioned to Taylor in private DM already, I'll skip this review today, but I may review this in hindsight when I'm back in office next week.

File: examples/nextjs-dashboard/src/components/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        }
+
+        return (
+          <Cursor
+            key={`cursor-${connectionId}`}
+            color={`rgb(${
+              COLORS_PRESENCE[connectionId % COLORS_PRESENCE.length]
+            }`}
+            x={(presence.cursor as CursorPosition).x * boundingRect.width}
+            y={(presence.cursor as CursorPosition).y * boundingRect.height}
```
Comment (CTNicholas) [inline examples/nextjs-dashboard/src/components/Card.tsx:99]: We can remove `as` here

File: examples/nextjs-dashboard/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+  const handleLegendPointerLeave = () => {
+    updateMyPresence({
+      selectedDataset: null,
+    });
+  };
+
+  const isDatasetSelected = (cardId: string, dataKey: DataKey<string>) => {
+    if (
+      (myPresence.selectedDataset as SelectedDataset)?.cardId === cardId &&
+      (myPresence.selectedDataset as SelectedDataset)?.dataKey === dataKey
```
Comment (CTNicholas) [inline examples/nextjs-dashboard/pages/index.tsx:63]: We can remove `as` here (see my Slack DM)

File: examples/nextjs-dashboard/pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    ) {
+      return true;
+    }
+
+    for (const [, selectedDataset] of others) {
+      if (
+        (selectedDataset as SelectedDataset)?.cardId === cardId &&
+        (selectedDataset as SelectedDataset)?.dataKey === dataKey
```
Comment (CTNicholas) [inline examples/nextjs-dashboard/pages/index.tsx:71]: We can remove `as` here

File: examples/nextjs-dashboard/pages/_app.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import "./globals.css";
+import Example from "./index";
+
+function App({ Component, pageProps }: AppProps) {
+  const roomId = useOverrideRoomId("nextjs-dashboard");
+  return (
+    <RoomProvider
+      id={roomId}
+      initialPresence={{
+        selectedDataset: { cardId: null, dataKey: null },
```
Comment (CTNicholas) [inline examples/nextjs-dashboard/pages/_app.tsx:15]:
```suggestion
        selectedDataset: null,
```

We can make this `null` initially (see I changed the type in `liveblocks.config.ts`)

File: examples/nextjs-dashboard/pages/index.tsx [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+} from "../src/data";
+import { useMyPresence, useOthersMapped } from "../src/liveblocks.config";
+import styles from "./index.module.css";
+import Header from "../src/components/Header";
+import Card from "../src/components/Card";
+
+type SelectedDataset = {
+  cardId: string;
+  dataKey: DataKey<string>;
+};
```
Comment (CTNicholas) [inline examples/nextjs-dashboard/pages/index.tsx:30]:
```suggestion
```

After removing `as`, we no longer need this 😄

File: examples/nextjs-dashboard/src/liveblocks.config.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { createClient } from "@liveblocks/client";
+import { createRoomContext } from "@liveblocks/react";
+
+const client = createClient({
+  publicApiKey: process.env.NEXT_PUBLIC_LIVEBLOCKS_PUBLIC_KEY!,
+});
+
+export type CursorPosition = {
```
Comment (CTNicholas) [inline examples/nextjs-dashboard/src/liveblocks.config.ts:8]:
```suggestion
type CursorPosition = {
```
After changes we don't need to export this anymore

File: examples/nextjs-dashboard/src/liveblocks.config.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  x: number;
+  y: number;
+};
+
+export type Presence = {
```
Comment (CTNicholas) [inline examples/nextjs-dashboard/src/liveblocks.config.ts:13]:
```suggestion
type Presence = {
```
After changes we don't need to export this anymore

**Review by CTNicholas — CHANGES_REQUESTED** [review]
Great work, basically just a couple of changes to the types and we're there 😄 

I've send a short video message on Slack with a proper explanation about the improvements we can make! Would advise taking a look before checking the review

**Review by CTNicholas — APPROVED** [review]
Good to go!

**Pull request merged**